stl/inc/regex

```diff
@@ -1270,17 +1270,40 @@ struct _Buf { // character buffer
     }
 
     template <class _FwdIt>
-    void _Insert(_FwdIt _First, _FwdIt _Last) { // append multiple characters
+    void _Insert2(_FwdIt _First, _FwdIt _Last) { // append multiple characters
         while (_First != _Last) {
-            _Insert(*_First++);
+            _Insert2(*_First++);
         }
     }
 
 private:
-    void _Expand(unsigned int _Len) { // expand buffer to hold _Len characters
-        _Elem* _Tmp = static_cast<_Elem*>(_CSTD realloc(_Chrs, _Get_size_of_n<sizeof(_Elem)>(_Len)));
+    unsigned int _Calculate_expansion(unsigned int _Increase) const {
+        constexpr size_t _Max_size_t     = static_cast<size_t>(-1) / sizeof(_Elem);
+        constexpr unsigned int _Max_uint = static_cast<unsigned int>(-1);
+        constexpr unsigned int _Max      = _Max_size_t < _Max_uint ? static_cast<unsigned int>(_Max_size_t) : _Max_uint;
+
+        if (_Increase < _Buf_incr) {
+            _Increase = _Buf_incr;

```

Comment:
Suggestion: As this is now the only usage of `_Buf_incr`, let's move the constant here, following our modern practice.